plugin/src/main/java/org/opensearch/ml/action/connector/DeleteConnectorTransportAction.java

```diff
@@ -117,22 +118,41 @@ private void handleConnectorAccessValidationFailure(String connectorId, Exceptio
 
     private void checkForModelsUsingConnector(String connectorId, String tenantId, ActionListener<DeleteResponse> actionListener) {
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            SearchRequest searchRequest = new SearchRequest(ML_MODEL_INDEX);
             SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
             sourceBuilder.query(QueryBuilders.matchQuery(MLModel.CONNECTOR_ID_FIELD, connectorId));
             if (mlFeatureEnabledSetting.isMultiTenancyEnabled()) {
                 sourceBuilder.query(QueryBuilders.matchQuery(TENANT_ID, tenantId));
             }
-            searchRequest.source(sourceBuilder);
-            // TODO: Use SDK client not client.
-            client.search(searchRequest, ActionListener.runBefore(ActionListener.wrap(searchResponse -> {
-                SearchHit[] searchHits = searchResponse.getHits().getHits();
-                if (searchHits.length == 0) {
-                    deleteConnector(connectorId, actionListener);
-                } else {
-                    handleModelsUsingConnector(searchHits, connectorId, actionListener);
-                }
-            }, e -> handleSearchFailure(connectorId, e, actionListener)), context::restore));
+
+            SearchDataObjectRequest searchDataObjectRequest = new SearchDataObjectRequest.Builder()
+                .indices(ML_MODEL_INDEX)
+                .searchSourceBuilder(sourceBuilder)
```

Comment:
I think we should also add tenant id in the sourceBuilder?

Reply:
Yes, have you already added it to all the others?  I lost track with the multiple PRs at the same time.

Reply:
No, I haven't made any changes with search functionalities yet.

Reply:
I think @arjunkumargiri is adding tenant id in https://github.com/opensearch-project/ml-commons/pull/2553/files

It's not currently in of the other requests; probably needs to be its own standalone field.

Reply:
yeah, I'll address that accordingly after that PR is being merged.